### ISSUE TITLE
[BUG]  marks.com supercookie comes back even after deleted

### ISSUE THREAD

**Issue by JohnnyCookie** [mrdokenny/Cookie-AutoDelete#1046]
**Describe the bug** marks.com supercookie

I visited marks.com a long time ago, probably 6-10 months ago, and this whole time, I keep seeing CAD delete it's cookies on every auto log. I've blocked it in my firefox browser settings - it keeps coming. This is the absolute most psychopathic cookie I've ever experienced. marks.com is the website for a retail store that sells clothing for blue collar workers, legitimate store - except they decided to use an ultra-aggressive cookie

**To Reproduce** visit marks.com

1. Go to 'marks.com'
2. Click on 'any part of the website'
3. Close the tab so CAD could auto delete  
4. See error of marks.com appearing in all future CAD logs

**Expected Behavior** MARKS.COM cookie will keep coming back

After visiting marks.com, leaving it, then allowing CAD to auto delete the website's cookies - you'll see that all future CAD logs will show the removal of marks.com cookies, long after you have already deleted their cookies, and this will repeat infinitely.

**Screenshots** 
None

**Your System Info** 

- OS: Windows 10 Home 
- Browser Info: Firefox 85.0.2 
- CookieAutoDelete Version: 3.6.0

**Comment by kennethtran93** [comment]
This is hard to say whether it is a CAD bug or a browser bug, or their server issue.  On my firefox (85.0), it works fine - only mentions marks.com once.

![image](https://user-images.githubusercontent.com/6724477/107600979-391a4e80-6bda-11eb-97a2-ef8b04d983a5.png)

I'll try and provide some support here, starting with a couple questions:

- What is your first party isolation setting value?
- what is your _dynamic_ first party isolation setting value?
- Do you have any additional site data cleanup options enabled?
- Are you using containers?
- Do you have any expressions configured for that site?  With an initial search available through the settings -> list of expressions page, try search the site up and see if anything comes up.
- What details shows up in the cleanup log for that site (a screenshot if possible)?
- This is a random thought, but did you block the site's cookie before or after the cookies were removed?
- We have a makeshift debug mode if you want to check it out and see if there's an issue with CAD's logic flow.  It does print a lot of lines, but you could filter by the site to make it easier.

The way CAD works, is we query the browser for all cookies, then filters them through the list of expressions, then passes the cookies/site data to be removed back to the browser to handle.

**Comment by neroux** [comment]
Possibly a duplicate of #807

**Comment by kennethtran93** [comment]
Maybe, but still waiting on the op's response.

**Comment by Hultan** [comment]
I am not sure if I should start a new issue or adding my comment to @JohnnyCookie :s issue, but I have a similar problem with now 20+ cookies that keeps "getting cleaned" over and over again. However, when I found the setting to enable container support just now, it is now at 40+ cookies that does not get cleaned!

I'll try to answer all of your questions above as best that I can:


* What is your first party isolation setting value?

    Are you talking about about:config -> privacy.firstparty.isolate? I have tried with it both on and off, and it makes no difference.

* what is your dynamic first party isolation setting value?

    I seached for **dynamic** and these five values showed up, I have not changed these values so they should be the default ones:

      javascript.options.dynamicImport | true |  
      layout.dynamic-reflow-roots.enabled | false | 
      layout.dynamic-toolbar-max-height | 0 |  
      privacy.dynamic_firstparty.limitForeign | false |  
      privacy.dynamic_firstparty.use_site | true |  

* Do you have any additional site data cleanup options enabled?

    If you are talking about the Privacy & Secutiry -> Cookies and Site Data section, than no. Nothing else that I know of...

* Are you using containers?

    Yes, I am using Multi-Account Containers, and as I mentioned above, when I checked the Container support option in CAD settings I am now up to 40+ cookies that are cleaned over and over again.

* Do you have any expressions configured for that site? With an initial search available through the settings -> list of expressions page, try search the site up and see if anything comes up.

    No, no matches for any of the original sites (before I checked Container support in CAD settings).

* What details shows up in the cleanup log for that site (a screenshot if possible)?

    Image from BEFORE I activated container support in CAD settings:

    [Details image](https://softteam.se/cookies.png)

* This is a random thought, but did you block the site's cookie before or after the cookies were removed?
    
    I do not understand the question, I did not white/gray-list them, so the cookies should be deleted, right?

* We have a makeshift debug mode if you want to check it out and see if there's an issue with CAD's logic flow. It does print a lot of lines, but you could filter by the site to make it easier.

     Please let me know how to activate it, and I'll give it a try if I can.

**Comment by kennethtran93** [comment]
@Hultan What is your Enhanced Tracking protection set to under Privacy & Security `about:preferences#privacy`?

**Comment by JohnnyCookie** [comment]
I just checked my logs - I can't see anything about marks.com in the logs.

I show in my CAD settings "No expressions defined."

    What is your first party isolation setting value? don't know what this is, i looked for it
    what is your dynamic first party isolation setting value?
    Do you have any additional site data cleanup options enabled? default i believe
    Are you using containers? i dont think so
    Do you have any expressions configured for that site? With an initial search available through the settings -> list of expressions page, try search the site up and see if anything comes up. NO
    What details shows up in the cleanup log for that site (a screenshot if possible)? **currently doesn't show up in the logs**
    This is a random thought, but did you block the site's cookie before or after the cookies were removed? **assumingly after it was removed, hard question because it kept showing up in the logs**
    We have a makeshift debug mode if you want to check it out and see if there's an issue with CAD's logic flow. It does print a lot of lines, but you could filter by the site to make it easier.


I recently updated windows 10 completely, and that made CAD start working again, prior to the windows update it stopped cleaning automatically. I also updated all my drivers - who knows if these factors helped or not. I'm also using the latest firefox 87.0
Another possibility - I'm the type of person who keeps 30-90+ tabs open at once, and I always restart the browser with the old tabs open. Recently, I started a new firefox session, after closing all the old tabs.

Props to you guys for staying on this, I remembered this thread a few days ago and figured it was too late and you would have shut down the thread by now

**Comment by kennethtran93** [comment]
@JohnnyCookie Not sure if Windows 10 updates have anything to do with this.  There is a possibility that you had one of those tabs in one of your many tabs.  As long as your privacy setting is not set to strict or your cookie behaviour is set to anything but 'cross-site cookies', then this extension should work fine for majority.

**Comment by JohnnyCookie** [comment]
The thing is I definitely didn't have a tab open for marks.com. Maybe I had an open tab that previously visited marks.com but was now at another website. I can't confirm that for sure - but that's still pretty messed up. I'm not an expert but I get a feeling it's related to the old firefox session..
The windows 10 update might not have anything to do with this problem cookie, but all I know is CAD stopped doing automatic updates, and once the windows 10 update was completed, all of a sudden CAD started working again

**Comment by Hultan** [comment]
Good morning, my enhanced tracking protection is **strict**. Does that influence how CAD works? What setting do you recommend, I am kind of an "extreme" privacy guy, but unfortunately I don't know too much about the different settings so I just go with whatever the latest YouTube-video recommends :-)

**Comment by kennethtran93** [comment]
@Hultan Unforunately strict mode is affecting all cookie webextensions at this time.  This is a firefox issue.  If you just want the previous strict mode (pre-86), use custom and set the cookie behavior to 'cross-site _tracking_ cookies'  See #807.  Unfortunately, all cookies created during the usage of strict mode from 86+ cannot be removed by us at this time.  Existing cookies should still be removable, but cannot be guaranteed.

**Comment by kennethtran93** [comment]
Closing this issue as OP appears to have resolved it, and the other issue belongs with the strict mode / cross-site cookies / first party dynamic issue in #807.

**Comment by Hultan** [comment]
Thank you, that solved the issue for me, I will use **custom** from now on...
